dxr/plugins/rust/tests/test_mods/test_mods.py

```diff
@@ -0,0 +1,120 @@
+from dxr.testing import DxrInstanceTestCase
+
+class ModTests(DxrInstanceTestCase):
+    def test_mod_def(self):
+        self.found_line_eq('module:mod1', "mod <b>mod1</b> {", 3)
+        self.found_line_eq('module:mod111', "pub mod <b>mod111</b> {", 5)
+        self.found_lines_eq('module:mod3',
+                            [("pub mod <b>mod3</b> {}", 13),
+                             ("pub mod <b>mod3</b> {}", 21)])
+
+    def test_mod_def_qual(self):
+        self.found_line_eq('+module:test::mod1', "mod <b>mod1</b> {", 3)
+        self.found_line_eq('+module:test::mod1::mod11::mod111', "pub mod <b>mod111</b> {", 5)
+        self.found_line_eq('+module:test::mod1::mod12::mod3', "pub mod <b>mod3</b> {}", 13)
+        self.found_line_eq('+module:test::mod2::mod3', "pub mod <b>mod3</b> {}", 21)
+
+    def test_mod_ref(self):
+        self.found_lines_eq('module-ref:mod1',
```

Comment:
As above, I wouldn't mind a comment (or a separate test) for each of these assertions so I can understand at a glance what unique thing it's testing.